metafix/src/main/java/org/metafacture/metafix/FixPath.java

```diff
@@ -0,0 +1,381 @@
+    }
+
+    /*package-private*/ FixPath(final String[] path) {
+        this.path = path;
+    }
+
+    public Value findIn(final Hash hash) {
+        final String field = path[0];
+        if (field.equals(ASTERISK)) {
+            // TODO: search in all elements of value.asHash()?
+            return new FixPath(tail(path)).findIn(hash);
+        }
+        return path.length == 1 || !hash.containsField(field) ? hash.get(field) :
+            findNested(hash, field, tail(path));
+    }
+
+    /*package-private*/ Value findIn(final Value.Array array) {
+        final Value result;
+        if (path.length > 0) {
+            final String currentSegment = path[0];
+            if (currentSegment.equals(ASTERISK)) {
+                result = Value.newArray(a -> array.forEach(v -> a.add(findInValue(v, tail(path)))));
+            }
+            else if (Value.isNumber(currentSegment)) {
+                final int index = Integer.parseInt(currentSegment) - 1; // TODO: 0-based Catmandu vs. 1-based Metafacture
+                if (index >= 0 && index < array.size()) {
+                    result = findInValue(array.get(index), tail(path));
+                }
+                else {
+                    result = null;
+                }
+            }
+            // TODO: WDCD? copy_field('your.name','author[].name'), where name is an array
+            else {
+                result = Value.newArray(a -> array.forEach(v -> a.add(findInValue(v, path))));
+            }
+        }
+        else {
+            result = new Value(array);
+        }
+        return result;
+    }
+
+    private Value findInValue(final Value value, final String[] p) {
+        // TODO: move impl into enum elements, here call only value.find
+        return value == null ? null : value.extractType((m, c) -> m
+                .ifArray(a -> c.accept(new FixPath(p).findIn(a)))
+                .ifHash(h -> c.accept(new FixPath(p).findIn(h)))
+                .orElse(c)
+                );
+    }
+
+    public Value replaceIn(final Hash hash, final String newValue) {
+        return new FixPath(path).insertInto(hash, InsertMode.REPLACE, new Value(newValue));
+    }
+
+    public Value appendIn(final Hash hash, final String newValue) {
+        return new FixPath(path).insertInto(hash, InsertMode.APPEND, new Value(newValue));
+    }
+
+    @Override
+    public String toString() {
+        return Arrays.asList(path).toString();
+    }
+
+    /*package-private*/ void transformIn(final Hash hash, final BiConsumer<TypeMatcher, Consumer<Value>> consumer) {
+        final Value oldValue = findIn(hash);
+
+        if (oldValue != null) {
+            final Value newValue = oldValue.extractType(consumer);
+
+            if (newValue != null) {
+                new FixPath(path).insertInto(hash, InsertMode.REPLACE, newValue);
+            }
+        }
+    }
+
+    /*package-private*/ void transformIn(final Hash hash, final UnaryOperator<String> operator) {
```

Comment:
Must be `public`. Currently needed in Limetrans. Or what is the intended replacement for `Record.transformFields(List<String>, UnaryOperator<String>)`?